projects/solar/src/test_data_tx.c

```diff
@@ -0,0 +1,93 @@
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "data_store.h"
+#include "data_tx.h"
+#include "event_queue.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ms_test_helpers.h"
+#include "soft_timer.h"
+#include "solar_events.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+int count = -1;                 // count of can msgs sent, incremented in rx_handler
+bool changeable_is_set = true;  // value that tells data_tx if data points are set
+static CanMessage s_can_msg;    // variable used to hold can msgs and verify values
+
+bool TEST_MOCK(data_store_get_is_set)(DataPoint data_point, bool *is_set) {
+  return changeable_is_set;
+}
+
+StatusCode TEST_MOCK(data_store_get)(DataPoint data_point, uint32_t *value) {
+  *value = data_point;
+  return STATUS_CODE_OK;
+}
+
+typedef enum { TEST_DATA_TX_EVENT_CAN_TX = 0, TEST_DATA_TX_EVENT_CAN_RX } TestDataTxEvent;
```

Comment:
To prevent clang-format from putting your enums (or structs/unions) on one line, you can put `//` after each line, like
```c
typedef enum {
  TEST_DATA_TX_EVENT_CAN_TX = 0,  //
  TEST_DATA_TX_EVENT_CAN_RX,      //
} TestDataTxEvent;
```
`make format` will automatically align the comments for you.

Also, this should have a `TEST_DATA_TX_EVENT_CAN_FAULT` member for the fault CAN event that the settings needs, and it should end with a `NUM_TEST_DATA_TX_CAN_EVENTS` member.